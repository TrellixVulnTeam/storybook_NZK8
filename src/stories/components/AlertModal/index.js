// @flow

import React from 'react';
import Modal, { ModalHeader, ModalBody, ModalFooter, type ModalHOCProps } from 'components/Modal';
import Button from 'components/Button';
import { colors } from 'utils/StyleGuide';
import { Wrapper } from './styled';

export type AlertModalProps = {
  header: string,
  body: string,
  maxWidth?: number,
  padding?: number,
  onConfirm: () => void | Promise<any>,
  confirmLabel?: string,
  onDeny?: () => void | Promise<any>,
  denyLabel?: string,
  onCancel?: () => void,
  cancelLabel?: string,
};

type Props = AlertModalProps & ModalHOCProps;

export default (props: Props) => {
  return (
    <Modal maxWidth={props.maxWidth || 540} padding={props.padding || 40}>
      <Wrapper>
        <ModalHeader
          onClose={props.onCancel || props.onClose}
          header={props.header}
          fontSize={18}
          fontWeight="600"
          color={colors.black}
          maxWidth={props.maxWidth || 540}
          padding={props.padding || 40}
          closeIconFill={colors.silver}
        />
        <ModalBody withPadding>{props.body}</ModalBody>
        <ModalFooter>
          {props.cancelLabel ? (
            <Button theme="cancel" size="xlg" onClick={props.onCancel || props.onClose}>
              {props.cancelLabel || props.t('generic.cancel')}
            </Button>
          ) : (
            <div />
          )}
          {props.denyLabel && props.onDeny ? (
            <Button theme="red" pill onClick={props.onDeny}>
              {props.denyLabel}
            </Button>
          ) : (
            <div />
          )}
          {props.confirmLabel ? (
            <Button theme="red" size="xxl" onClick={props.onConfirm}>
              {props.confirmLabel || props.t('generic.confirm')}
            </Button>
          ) : (
            <div />
          )}
        </ModalFooter>
      </Wrapper>
    </Modal>
  );
};
