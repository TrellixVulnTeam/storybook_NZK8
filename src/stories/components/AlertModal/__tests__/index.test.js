import React from 'react';
import { mount } from 'enzyme';
import AlertModal from 'components/AlertModal/index';

describe('<AlertModal />', () => {
  const header = 'Header';
  const body = 'Body';

  it('should render correctly with all params given', () => {
    const sut = mount(
      <AlertModal
        header={header}
        body={body}
        maxWidth={100}
        padding={100}
        confirmLabel="ConfirmLabel"
        denyLabel="DenyLabel"
        cancelLabel="CancelLabel"
        onDeny={jest.fn()}
        onCancel={jest.fn()}
        onConfirm={jest.fn()}
      />,
    );

    expect(sut).toMatchSnapshot();
  });

  it('should render correctly with only madatory params given', () => {
    const sut = mount(<AlertModal header={header} body={body} onConfirm={jest.fn()} />);

    expect(sut).toMatchSnapshot();
  });

  it('should render correctly with specific params given', () => {
    const sut = mount(
      <AlertModal
        header={header}
        body={body}
        onConfirm={jest.fn()}
        onClose={jest.fn()}
        cancelLabel="CancelLabel"
      />,
    );

    expect(sut).toMatchSnapshot();
  });
});
