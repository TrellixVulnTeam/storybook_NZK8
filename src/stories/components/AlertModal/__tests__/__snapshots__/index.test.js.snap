// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AlertModal /> should render correctly with all params given 1`] = `
<_default
  body="Body"
  cancelLabel="CancelLabel"
  confirmLabel="ConfirmLabel"
  denyLabel="DenyLabel"
  header="Header"
  maxWidth={100}
  onCancel={[MockFunction]}
  onConfirm={[MockFunction]}
  onDeny={[MockFunction]}
  padding={100}
>
  <Modal
    maxWidth={100}
    padding={100}
  >
    <styled__Background>
      <div
        className="styled__Background-sc-1l0ief0-0 iqpWHT"
      >
        <styled__ModalWrapper
          maxWidth={100}
          padding={100}
        >
          <div
            className="styled__ModalWrapper-sc-1l0ief0-1 eCEmXl"
          >
            <styled__Wrapper
              key=".0"
              maxWidth={100}
              padding={100}
            >
              <div
                className="styled__Wrapper-sc-1g6plfk-0 QCFGq"
              >
                <_default
                  closeIconFill="#b7bbbf"
                  color="#000000"
                  fontSize={18}
                  fontWeight="600"
                  header="Header"
                  maxWidth={100}
                  onClose={[MockFunction]}
                  padding={100}
                >
                  <styled__Wrapper>
                    <div
                      className="styled__Wrapper-sc-1mqwacl-0 kciYOf"
                    >
                      <styled__HeaderWrapper
                        color="#000000"
                        fontSize={18}
                        fontWeight="600"
                      >
                        <div
                          className="styled__HeaderWrapper-sc-1mqwacl-1 hMQzvg"
                          color="#000000"
                          fontSize={18}
                          fontWeight="600"
                        >
                          <div>
                            Header
                          </div>
                          <styled__CloseWrapper
                            color="#b7bbbf"
                            onClick={[Function]}
                          >
                            <div
                              className="styled__CloseWrapper-sc-1mqwacl-2 iDPARE"
                              color="#b7bbbf"
                              onClick={[Function]}
                            >
                              <_default
                                id="modal-close"
                                value="Close"
                              >
                                <styled__Wrapper
                                  data-for="modal-close"
                                  data-tip="Close"
                                  id="modal-close"
                                  value="Close"
                                >
                                  <div
                                    className="styled__Wrapper-funpqb-0 cznzQQ"
                                    data-for="modal-close"
                                    data-tip="Close"
                                    id="modal-close"
                                    value="Close"
                                  >
                                    <styled__CloseIcon>
                                      <test-file-stub
                                        className="styled__CloseIcon-sc-1mqwacl-3 daqvoq"
                                      />
                                    </styled__CloseIcon>
                                    <styled__TooltipWrapper
                                      display="true"
                                    >
                                      <div
                                        className="styled__TooltipWrapper-funpqb-1 gQtIVd"
                                        display="true"
                                      >
                                        <ReactTooltip
                                          clickable={false}
                                          delayShow={300}
                                          effect="solid"
                                          getContent={[Function]}
                                          html={true}
                                          id="modal-close"
                                          insecure={true}
                                          multiline={true}
                                          place="top"
                                          resizeHide={true}
                                          type="dark"
                                          wrapper="div"
                                        >
                                          <div
                                            className="__react_component_tooltip place-top type-dark"
                                            data-id="tooltip"
                                            id="modal-close"
                                          >
                                            null
                                          </div>
                                        </ReactTooltip>
                                      </div>
                                    </styled__TooltipWrapper>
                                  </div>
                                </styled__Wrapper>
                              </_default>
                            </div>
                          </styled__CloseWrapper>
                        </div>
                      </styled__HeaderWrapper>
                    </div>
                  </styled__Wrapper>
                </_default>
                <ModalBody
                  withPadding={true}
                >
                  <div
                    className="ModalBody-sc-1bmeib-0 cKfmGH"
                  >
                    Body
                  </div>
                </ModalBody>
                <ModalFooter>
                  <div
                    className="ModalFooter-sc-1s2gfs4-0 jOWMLs"
                  >
                    <_default
                      onClick={[MockFunction]}
                      size="xlg"
                      theme="cancel"
                    >
                      <styled__Wrapper
                        onClick={[Function]}
                        size="xlg"
                        theme="cancel"
                        type="button"
                      >
                        <button
                          className="styled__Wrapper-sc-6m8pxr-0 kQsakw"
                          onClick={[Function]}
                          size="xlg"
                          type="button"
                        >
                          CancelLabel
                        </button>
                      </styled__Wrapper>
                    </_default>
                    <_default
                      onClick={[MockFunction]}
                      pill={true}
                      theme="red"
                    >
                      <styled__Wrapper
                        onClick={[Function]}
                        pill={true}
                        theme="red"
                        type="button"
                      >
                        <button
                          className="styled__Wrapper-sc-6m8pxr-0 dnowDK"
                          onClick={[Function]}
                          type="button"
                        >
                          DenyLabel
                        </button>
                      </styled__Wrapper>
                    </_default>
                    <_default
                      onClick={[MockFunction]}
                      size="xxl"
                      theme="red"
                    >
                      <styled__Wrapper
                        onClick={[Function]}
                        size="xxl"
                        theme="red"
                        type="button"
                      >
                        <button
                          className="styled__Wrapper-sc-6m8pxr-0 ksGpwH"
                          onClick={[Function]}
                          size="xxl"
                          type="button"
                        >
                          ConfirmLabel
                        </button>
                      </styled__Wrapper>
                    </_default>
                  </div>
                </ModalFooter>
              </div>
            </styled__Wrapper>
          </div>
        </styled__ModalWrapper>
      </div>
    </styled__Background>
  </Modal>
</_default>
`;

exports[`<AlertModal /> should render correctly with only madatory params given 1`] = `
<_default
  body="Body"
  header="Header"
  onConfirm={[MockFunction]}
>
  <Modal
    maxWidth={540}
    padding={40}
  >
    <styled__Background>
      <div
        className="styled__Background-sc-1l0ief0-0 iqpWHT"
      >
        <styled__ModalWrapper
          maxWidth={540}
          padding={40}
        >
          <div
            className="styled__ModalWrapper-sc-1l0ief0-1 hAvnrx"
          >
            <styled__Wrapper
              key=".0"
              maxWidth={540}
              padding={40}
            >
              <div
                className="styled__Wrapper-sc-1g6plfk-0 QCFGq"
              >
                <_default
                  closeIconFill="#b7bbbf"
                  color="#000000"
                  fontSize={18}
                  fontWeight="600"
                  header="Header"
                  maxWidth={540}
                  padding={40}
                >
                  <styled__Wrapper>
                    <div
                      className="styled__Wrapper-sc-1mqwacl-0 kciYOf"
                    >
                      <styled__HeaderWrapper
                        color="#000000"
                        fontSize={18}
                        fontWeight="600"
                      >
                        <div
                          className="styled__HeaderWrapper-sc-1mqwacl-1 hMQzvg"
                          color="#000000"
                          fontSize={18}
                          fontWeight="600"
                        >
                          <div>
                            Header
                          </div>
                          <styled__CloseWrapper
                            color="#b7bbbf"
                            onClick={[Function]}
                          >
                            <div
                              className="styled__CloseWrapper-sc-1mqwacl-2 iDPARE"
                              color="#b7bbbf"
                              onClick={[Function]}
                            >
                              <_default
                                id="modal-close"
                                value="Close"
                              >
                                <styled__Wrapper
                                  data-for="modal-close"
                                  data-tip="Close"
                                  id="modal-close"
                                  value="Close"
                                >
                                  <div
                                    className="styled__Wrapper-funpqb-0 cznzQQ"
                                    data-for="modal-close"
                                    data-tip="Close"
                                    id="modal-close"
                                    value="Close"
                                  >
                                    <styled__CloseIcon>
                                      <test-file-stub
                                        className="styled__CloseIcon-sc-1mqwacl-3 daqvoq"
                                      />
                                    </styled__CloseIcon>
                                    <styled__TooltipWrapper
                                      display="true"
                                    >
                                      <div
                                        className="styled__TooltipWrapper-funpqb-1 gQtIVd"
                                        display="true"
                                      >
                                        <ReactTooltip
                                          clickable={false}
                                          delayShow={300}
                                          effect="solid"
                                          getContent={[Function]}
                                          html={true}
                                          id="modal-close"
                                          insecure={true}
                                          multiline={true}
                                          place="top"
                                          resizeHide={true}
                                          type="dark"
                                          wrapper="div"
                                        >
                                          <div
                                            className="__react_component_tooltip place-top type-dark"
                                            data-id="tooltip"
                                            id="modal-close"
                                          >
                                            null
                                          </div>
                                        </ReactTooltip>
                                      </div>
                                    </styled__TooltipWrapper>
                                  </div>
                                </styled__Wrapper>
                              </_default>
                            </div>
                          </styled__CloseWrapper>
                        </div>
                      </styled__HeaderWrapper>
                    </div>
                  </styled__Wrapper>
                </_default>
                <ModalBody
                  withPadding={true}
                >
                  <div
                    className="ModalBody-sc-1bmeib-0 cKfmGH"
                  >
                    Body
                  </div>
                </ModalBody>
                <ModalFooter>
                  <div
                    className="ModalFooter-sc-1s2gfs4-0 jOWMLs"
                  >
                    <div />
                    <div />
                    <div />
                  </div>
                </ModalFooter>
              </div>
            </styled__Wrapper>
          </div>
        </styled__ModalWrapper>
      </div>
    </styled__Background>
  </Modal>
</_default>
`;

exports[`<AlertModal /> should render correctly with specific params given 1`] = `
<_default
  body="Body"
  cancelLabel="CancelLabel"
  header="Header"
  onClose={[MockFunction]}
  onConfirm={[MockFunction]}
>
  <Modal
    maxWidth={540}
    padding={40}
  >
    <styled__Background>
      <div
        className="styled__Background-sc-1l0ief0-0 iqpWHT"
      >
        <styled__ModalWrapper
          maxWidth={540}
          padding={40}
        >
          <div
            className="styled__ModalWrapper-sc-1l0ief0-1 hAvnrx"
          >
            <styled__Wrapper
              key=".0"
              maxWidth={540}
              padding={40}
            >
              <div
                className="styled__Wrapper-sc-1g6plfk-0 QCFGq"
              >
                <_default
                  closeIconFill="#b7bbbf"
                  color="#000000"
                  fontSize={18}
                  fontWeight="600"
                  header="Header"
                  maxWidth={540}
                  onClose={[MockFunction]}
                  padding={40}
                >
                  <styled__Wrapper>
                    <div
                      className="styled__Wrapper-sc-1mqwacl-0 kciYOf"
                    >
                      <styled__HeaderWrapper
                        color="#000000"
                        fontSize={18}
                        fontWeight="600"
                      >
                        <div
                          className="styled__HeaderWrapper-sc-1mqwacl-1 hMQzvg"
                          color="#000000"
                          fontSize={18}
                          fontWeight="600"
                        >
                          <div>
                            Header
                          </div>
                          <styled__CloseWrapper
                            color="#b7bbbf"
                            onClick={[Function]}
                          >
                            <div
                              className="styled__CloseWrapper-sc-1mqwacl-2 iDPARE"
                              color="#b7bbbf"
                              onClick={[Function]}
                            >
                              <_default
                                id="modal-close"
                                value="Close"
                              >
                                <styled__Wrapper
                                  data-for="modal-close"
                                  data-tip="Close"
                                  id="modal-close"
                                  value="Close"
                                >
                                  <div
                                    className="styled__Wrapper-funpqb-0 cznzQQ"
                                    data-for="modal-close"
                                    data-tip="Close"
                                    id="modal-close"
                                    value="Close"
                                  >
                                    <styled__CloseIcon>
                                      <test-file-stub
                                        className="styled__CloseIcon-sc-1mqwacl-3 daqvoq"
                                      />
                                    </styled__CloseIcon>
                                    <styled__TooltipWrapper
                                      display="true"
                                    >
                                      <div
                                        className="styled__TooltipWrapper-funpqb-1 gQtIVd"
                                        display="true"
                                      >
                                        <ReactTooltip
                                          clickable={false}
                                          delayShow={300}
                                          effect="solid"
                                          getContent={[Function]}
                                          html={true}
                                          id="modal-close"
                                          insecure={true}
                                          multiline={true}
                                          place="top"
                                          resizeHide={true}
                                          type="dark"
                                          wrapper="div"
                                        >
                                          <div
                                            className="__react_component_tooltip place-top type-dark"
                                            data-id="tooltip"
                                            id="modal-close"
                                          >
                                            null
                                          </div>
                                        </ReactTooltip>
                                      </div>
                                    </styled__TooltipWrapper>
                                  </div>
                                </styled__Wrapper>
                              </_default>
                            </div>
                          </styled__CloseWrapper>
                        </div>
                      </styled__HeaderWrapper>
                    </div>
                  </styled__Wrapper>
                </_default>
                <ModalBody
                  withPadding={true}
                >
                  <div
                    className="ModalBody-sc-1bmeib-0 cKfmGH"
                  >
                    Body
                  </div>
                </ModalBody>
                <ModalFooter>
                  <div
                    className="ModalFooter-sc-1s2gfs4-0 jOWMLs"
                  >
                    <_default
                      onClick={[MockFunction]}
                      size="xlg"
                      theme="cancel"
                    >
                      <styled__Wrapper
                        onClick={[Function]}
                        size="xlg"
                        theme="cancel"
                        type="button"
                      >
                        <button
                          className="styled__Wrapper-sc-6m8pxr-0 kQsakw"
                          onClick={[Function]}
                          size="xlg"
                          type="button"
                        >
                          CancelLabel
                        </button>
                      </styled__Wrapper>
                    </_default>
                    <div />
                    <div />
                  </div>
                </ModalFooter>
              </div>
            </styled__Wrapper>
          </div>
        </styled__ModalWrapper>
      </div>
    </styled__Background>
  </Modal>
</_default>
`;
